database/snowflake.js

```diff
@@ -0,0 +1,88 @@
+const snowflake = require('snowflake-sdk')
+const genericPool = require("generic-pool");
+const config = require('../config')
+const crypto = require('crypto')
+
+const factory = {
+  create: () => {
+      return new Promise((resolve, reject) => {
+          // Create Connection
+          const connection = snowflake.createConnection({
+            account: config.snowflake_account,
+            username: config.snowflake_user,
+            authenticator: 'SNOWFLAKE_JWT',
+            privateKey: crypto.createPrivateKey({
+              key: config.snowflake_private_key,
+              format: 'pem'
+            }).export({
+              format: 'pem',
+              type: 'pkcs8'
+            }),
+            database: config.snowflake_database,
+            warehouse: config.snowflake_warehouse,
+            clientSessionKeepAlive: true
+          });
+          // Try to connect to Snowflake, and check whether the connection was successful.
+          connection.connect((err, conn) => {
+              if (err) {
+                  console.error('Unable to connect: ' + err.message);
+                  reject(new Error(err.message));
+              } else {
+                  console.log('Successfully connected to Snowflake, ID:', conn.getId());
+                  resolve(conn);
+              }
+          });
+      });
+  },
+  destroy: (connection) => {
+      return new Promise((resolve, reject) => {
+          connection.destroy((err, conn) => {
+              if (err) {
+                  console.error('Unable to disconnect: ' + err.message);
+              } else {
+                  console.log('Disconnected connection with id: ' + conn.getId());
+              }
+              resolve(); // Always resolve for destroy
+          });
+      });
+  },
+  validate: (connection) => {
+      return new Promise((resolve, reject) => {
+          resolve(connection.isUp());
+      });
+  }
+};
+
+const opts = {
+  max: 100, // Maximum size of the pool
+  min: 2, // Minimum size of the pool,
+  testOnBorrow: false, // Validate connection before acquiring it
+  acquireTimeoutMillis: 60000, // Timeout to acquire connection
+  evictionRunIntervalMillis: 900000, // Check every 15 min for ideal connection
+  numTestsPerEvictionRun: 3, // Check only 3 connections every 15 min
+  idleTimeoutMillis: 10800000 // Evict only if connection is ideal for 3 hrs
```

Comment:
Typo:  Evict only if connection is ideal for 3 hrs --> Evict only if connection is idle for 3 hrs